src/client/instructureComponents/selectCompleteForm.js

```diff
@@ -0,0 +1,215 @@
+    // Input value is from highlighted option, not user input
+    // clear input, reset options
+    if (highlightedOptionId) {
+      if (inputValue === getOptionById(highlightedOptionId).label) {
+        setInputValue('');
+        setFilteredOptions(filterOptions(''));
+      }
+    }
+  };
+
+  const handleHideOptions = () => {
+    setIsShowingOptions(false);
+    setHighlightedOptionId(null);
+    matchValue();
+  };
+
+  const handleBlur = () => {
+    setHighlightedOptionId(null);
+  };
+
+  const handleHighlightOption = (event, id) => {
+    event.persist();
+    const option = getOptionById(id);
+    if (!option) return; // Prevent highlighting of empty option
+    setHighlightedOptionId(id);
+    setInputValue(event.type === 'keydown' ? option.label : inputValue);
+  };
+
+  const handleSelectOption = (event, id) => {
+    const option = getOptionById(id);
+    if (!option) return; // Prevent selecting of empty option
+    setSelectedOptionId(id);
+    setInputValue(option.label);
+    setIsShowingOptions(false);
+    setFilteredOptions(options);
+    onSelect(option);
+    //
+  };
+
+  const handleInputChange = event => {
+    const { value } = event.target;
+    const newOptions = filterOptions(value);
+    let newHighlightedOptionId = null;
+    if (!isGroup) {
+      newHighlightedOptionId = newOptions.length > 0 ? newOptions[0].id : null;
+    } else {
+      const groups = Object.keys(newOptions);
+      newHighlightedOptionId =
+        groups.length > 0 ? newOptions[groups[0]][0].id : null;
+    }
+    setInputValue(value);
+    setFilteredOptions(newOptions);
+    setHighlightedOptionId(newHighlightedOptionId);
+    setIsShowingOptions(true);
+    setSelectedOptionId(value === '' ? null : selectedOptionId);
+  };
+
+  const renderGroup = () =>
+    Object.keys(filteredOptions).map((key, index) => (
+      <Select.Group key={index} renderLabel={key}>
+        {filteredOptions[key].map(option => (
+          <Select.Option
+            key={option.id}
+            id={option.id}
+            isHighlighted={option.id === highlightedOptionId}
+            isSelected={option.id === selectedOptionId}
+          >
+            {option.label}
+          </Select.Option>
+        ))}
+      </Select.Group>
+    ));
+
+  const renderOptions = () => (
+    <>
+      {filteredOptions.length > 0 ? (
+        filteredOptions.map(option => (
+          <Select.Option
+            id={option.id}
+            key={option.id}
+            isHighlighted={option.id === highlightedOptionId}
+            isSelected={option.id === selectedOptionId}
+            isDisabled={option.disabled}
+          >
+            {!option.disabled ? option.label : `${option.label} (unavailable)`}
+          </Select.Option>
+        ))
+      ) : (
+        <Select.Option id="empty-option" key="empty-option">
+          ---
+        </Select.Option>
+      )}
+    </>
+  );
+
+  return (
+    <div>
+      <Select
+        renderLabel="Terms auto complete"
```

Comment:
@tatikamsetti This is called SelectCompleteForm, but it appears to only do auto complete for terms? Shouldn't this be called something like AutomCompleteTermForm?